source/javascripts/components/StacksAndMachine/MachineTypeSelector.tsx

```diff
@@ -1,36 +1,132 @@
 import { MachineTypeWithValue } from '@/core/services/StackAndMachineService';
 
+export const getMachineTypePromotionTexts = (isTrialMode: boolean) => {
+  return isTrialMode
+    ? {
+        availableText: 'Available during the trial',
+        promotedText: 'Available on paid plans',
+        toggleTipText:
+          'Select your machine type for builds. Options vary by plan. If you need stronger machines during your trial, contact sales.',
+      }
+    : {
+        availableText: 'Available on your plan',
+        promotedText: 'Available on other plans',
+        toggleTipText:
+          'Select your machine type for builds. Options vary by plan. If you need stronger machines contact sales.',
+      };
+};
+
 type Props = {
-  options: MachineTypeOption[];
+  availableOptions: MachineTypeOption[];
   machineType: MachineTypeWithValue;
   onChange: (machineId: string) => void;
+  promotedOptions: MachineTypeOption[];
   isLoading: boolean;
   isDisabled: boolean;
   isInvalid: boolean;
+  isMachineTypePromotionTrialMode: boolean;
 };
 
-const MachineTypeSelector = ({ machineType, options, isLoading, isDisabled, isInvalid, onChange }: Props) => {
+const MachineTypeSelector = ({
+  machineType,
+  availableOptions,
+  isLoading,
+  isDisabled,
+  isInvalid,
+  isMachineTypePromotionTrialMode,
+  onChange,
+  promotedOptions,
+}: Props) => {
+  const machineTypePromotion = getMachineTypePromotionTexts(isMachineTypePromotionTrialMode);
+
+  const toggletip = (icon: ReactNode) => (
+    <Toggletip
+      button={{ href: 'https://bitrise.io/demo', label: 'Reach out to sales' }}
+      label={machineTypePromotion.toggleTipText}
+    >
+      {icon}
+    </Toggletip>
+  );
+
   return (
-    <Select
-      isRequired
-      label="Machine type"
-      isLoading={isLoading}
-      value={machineType.value}
-      isDisabled={isDisabled}
+    <Dropdown
+      disabled={isLoading || isDisabled}
       errorText={isInvalid ? 'Invalid machine type' : undefined}
       helperText={machineType.creditPerMinute ? `${machineType.creditPerMinute} credits/min` : undefined}
-      onChange={(e) => onChange(e.target.value)}
-      flex="1"
+      label="Machine type"
+      labelHelp={toggletip}
+      onChange={(e) => onChange(e.target.value as string)}
+      required
+      search={false}
+      value={machineType.value}
     >
-      {options.map(({ value, label }) => (
-        <option key={value} value={value}>
-          {label}
-        </option>
-      ))}
-    </Select>
+      {[
+        {
+          isEnabled: true,
+          label: machineTypePromotion.availableText,
+          options: availableOptions,
+        },
+        {
+          isEnabled: false,
+          label: machineTypePromotion.promotedText,
+          options: promotedOptions,
+        },
+      ].map(({ isEnabled, label, options }) => {
+        const machineOptions = options?.map(({ label: optionLabel, value, osId }) => {
```

Comment:
double-map, it's a strong signal, that you would want to create separate sub-components here

Reply:
Dropdown seems to not handle separate sub-components well - it shows the dropdown's opener "input" button empty. The best I could do is create a function instead of a separate sub-component.